src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -104,33 +106,125 @@ public static void runQueryBenchmarks(List<QueryBenchmark> queryBenchmarks, Stri
 
 		    for (int threads = benchmark.minThreads; threads <= benchmark.maxThreads; threads++) {
 		        QueryGenerator queryGenerator = new QueryGenerator(benchmark);
+		        List<SolrBenchQuery> queryResults = new Vector<>();
```

Comment:
Experimented a bit in [this change](https://github.com/fullstorydev/solr-bench/pull/46/files) to see whether we can get generated queries from `controllerExecutor#run` instead. 😊 

The proposed changes aim at:
1. Keep the query supplier focused on generating/supplying the queries, it might not be entirely obvious  of why query supplier having an list of `queryResults` as input param (and as mutable, which triggers other concerns such as concurrent access, logic flow etc). 
2. On the other end if `controllerExecutor#run` method returns a list of "result", that would be more straight forward to understand - these are the results obtained from executing all the supplied runnable/callables.

Reply:
Sounds good :-)

Reply:
https://github.com/fullstorydev/solr-bench/pull/46

I think once ^ this one is merged, I'll rebase this PR. :-)